addons/mail/static/src/messaging/entity/attachment/attachment_tests.js

```diff
@@ -0,0 +1,154 @@
+odoo.define('mail.messaging.entity.AttachmentTests', function (require) {
+'use strict';
+
+const {
+    afterEach: utilsAfterEach,
+    beforeEach: utilsBeforeEach,
+    pause,
+    start: utilsStart,
+} = require('mail.messaging.testUtils');
+
+QUnit.module('mail', {}, function () {
+QUnit.module('messaging', {}, function () {
+QUnit.module('entity', {}, function () {
+QUnit.module('Attachment', {
+    beforeEach() {
+        utilsBeforeEach(this);
+        this.start = async params => {
+            if (this.widget) {
+                this.widget.destroy();
+            }
+            let { env, widget } = await utilsStart(Object.assign({}, params, {
+                data: this.data,
+            }));
+            this.env = env;
+            this.widget = widget;
+        };
+    },
+    afterEach() {
+        utilsAfterEach(this);
+        this.env = undefined;
+        if (this.widget) {
```

Comment:
This is everywhere, but do we really need that condition? Is `this.widget` ever falsy? I feel like the test would fail if the widget is not correctly set?

Reply:
it's always truthy, but let's keep like this for the moment because I intend to refactor `start()` method so that it only sets up the environment (not discuss, messaging menu, etc..., which is why it has `widget` to destroy everything at the moment)

Reply:
This looks like the opposite of what I would want but we can check that after merge.